usr/src/boot/efi/libfdtutil/fixup.c

```diff
@@ -0,0 +1,114 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2025 Michael van der Westhuizen
+ */
+
+/*
+ * FDT fixups, to make things more predictable for the kernel.
+ */
+
+#include <stand.h>
+#include <libfdt.h>
+#include <fdt.h>
+
+#define	FDT_FIX_PHANDLE_SUCCESS	0
+#define	FDT_FIX_PHANDLE_AGAIN	1
+#define	FDT_FIX_PHANDLE_FAILURE	2
+
+
+static int
+i_fix_phandle(void *fdtp, int node)
```

Comment:
Can we insert a comment sort of like:

```c
/*
  * We give each node a phandle in the FDT so that illumos interfaces that look up nodes by identifier have an identifier available for all nodes.
  */
```

I've been wondering if we could and should instead use the node address for this (with a larger phandle_t/pnode_t), or even if we're bad people, just the low 32bits of the address, since it won't actually collide.

Reply:
Added a full comment in the file header explaining the fixups. The comment is forward-looking, and states that the FDT will be treated as read-only by the kernel.